src/sdk/base/EtherspotWalletAPI.ts

```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
+  // deinitData is prepared as bytes data made of the previous node address and the deinit data
+  // the deinit data is the data that is passed to the module to be uninstalled
+  async generateModuleDeInitData(moduleTypeId: MODULE_TYPE, module: string, deinitData: string): Promise<string> {
+
+    // this is applicable only for Executor and Validator modules
+    // if the module type is not Executor or Validator, throw an error
+    if (moduleTypeId !== MODULE_TYPE.EXECUTOR && moduleTypeId !== MODULE_TYPE.VALIDATOR) {
+      throw new Error("Unsupported module type");
+    }
+
+    // Get the previous address in the list
+    const previousAddress = await this.getPreviousAddressInSentinelList(module, moduleTypeId);
+
+    console.log(`Previous address: ${previousAddress}`);
+
+    // Prepare the deinit data
+    const deInitData = ethers.utils.defaultAbiCoder.encode(
+      ["address", "bytes"],
+      [previousAddress, deinitData]
+    );
+
+    return deInitData;
+  }
+
+  // function to get validators
+  async getAllValidators(): Promise<string[]> {
+    // Assuming getValidatorsPaginated is a contract function that can be called
+    // Initialize variables
+    let pageSize = 50;
+    let lastAddress = SENTINEL_ADDRESS;
+    let totalValidators: string[] = [];
+
+    // Pagination loop
+    let tempValidators: string[];
+    const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+
+    do {
+      // Fetch a page of validators
+      [tempValidators, lastAddress] = await accountContract.getValidatorPaginated(lastAddress, pageSize);
+
+      // Append validators to the total list
+      totalValidators = [...totalValidators, ...tempValidators];
+
+      // Break if it's the last page
+      if (tempValidators.length < pageSize || lastAddress === SENTINEL_ADDRESS || lastAddress === ethers.constants.AddressZero) {
+        break;
+      }
+    } while (true);
+
+    return totalValidators;
+  }
+
+  // function to get active hook
+  async getActiveHook(): Promise<string> {
+    const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+    return accountContract.getActiveHook();
+  }
+
+  async getFallbacks(): Promise<any[]> {
+    return [];
+  }
+
+  // function to club the response of getAllExecutors, getAllValidators and getActiveHook
+  // return should be a wrapper of tis way
+  // prepare a schema like above and return the response
+  async getAllModules(): Promise<ModuleInfo> {
+    const validators = await this.getAllValidators() || [];
+    const executors = await this.getAllExecutors() || [];
+    const hook = await this.getActiveHook() || "";
+    const fallbacks = await this.getFallbacks() || [];
+
+    return {
+      validators,
+      executors,
+      hook,
+      fallbacks
+    };
+  }
+
+
   async checkAccountAddress(address: string): Promise<void> {
+    console.log(`inside checkAccountAddress for: ${address}`);
```

Comment:
Please remove console.log

Reply:
oops forgot to push removed changes while raising PR. removed it now